packages/clay-core/src/tree-view/useMultipleSelection.tsx

```diff
@@ -25,17 +29,28 @@ export interface IMultipleSelectionState {
 	toggleSelection: (key: Key) => void;
 }
 
-export interface IMultipleSelectionProps extends IMultipleSelection {}
+export interface IMultipleSelectionProps<T>
+	extends IMultipleSelection,
+		ITreeProps<T>,
+		Pick<ICollectionProps<T>, 'items'> {
+	multipleSelection?: boolean;
+}
 
 type LayoutInfo = {
 	children: Set<Key>;
 	intermediate: boolean;
 	parentKey?: Key;
 };
 
-export function useMultipleSelection(
-	props: IMultipleSelectionProps
+export function useMultipleSelection<T>(
+	props: IMultipleSelectionProps<T>
 ): IMultipleSelectionState {
+	const multipleSelection = props.multipleSelection;
+
+	const nestedKey = props.nestedKey ?? 'children';
+
+	const tree = createImmutableTree(props.items ?? [], nestedKey);
```

Comment:
In this case I think we should consider calling this method only where it's being used, because otherwise the `tree` will always be created every time there's a new rendering is a bit unnecessary cost in this case here.

Reply:
I don't think creating the tree is that expensive, however I agree that we only need it where it's used

Reply:
Yeah in this case it's not necessarily because we're not doing any operations yet but the problem is more related that this method will create a new reference every time a new rendering happens and internally we just create a partial copy of the first level with destructuring but not much thing too.